.github/workflows/scripts/linux/build-dependencies.sh

```diff
@@ -0,0 +1,125 @@
+SRCDIR="$HOME"
+INSTALLDIR="$HOME/Depends"
+
+QT=6.3.0
+SDL=SDL2-2.0.22
+PATCHELF_VERSION=0.14.5
+
+export CC=gcc-10; export CXX=g++-10
+export CFLAGS="-I$INSTALLDIR/include -Os $CFLAGS"
+export CXXFLAGS="-I$INSTALLDIR/include -Os $CXXFLAGS"
+export LDFLAGS="-L$INSTALLDIR/lib $LDFLAGS"
+export PKG_CONFIG_PATH="$INSTALLDIR/lib/pkgconfig:$PKG_CONFIG_PATH"
+
+export Qt6_DIR="$HOME/Depends/lib/cmake/Qt6"
+export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$HOME/Depends/lib/
+
+cd "$SRCDIR"
+
+cat > SHASUMS <<EOF
+fe7cbf3127882e3fc7259a75a0cb585620272c51745d3852ab9dd87960697f2e  $SDL.tar.gz
+b9a46f2989322eb89fa4f6237e20836c57b455aa43a32545ea093b431d982f5c  patchelf-$PATCHELF_VERSION.tar.bz2
+b865aae43357f792b3b0a162899d9bf6a1393a55c4e5e4ede5316b157b1a0f99  qtbase-everywhere-src-$QT.tar.xz
+d294b029dc2b2d4f65da516fdc3b8088d32643eb7ff77db135a8b9ce904caa37  qtdeclarative-everywhere-src-$QT.tar.xz
+3164504d7e3f640439308235739b112605ab5fc9cc517ca0b28f9fb93a8db0e3  qtsvg-everywhere-src-$QT.tar.xz
+fce94688ea925782a2879347584991f854630daadba6c52aed6d93e33cd0b19c  qttools-everywhere-src-$QT.tar.xz
+e4dd4ef892a34a9514a19238f189a33ed85c76f31dcad6599ced93b1e33440b3  qttranslations-everywhere-src-$QT.tar.xz
+e7b567f6e43ffc5918d4aa825ce1eced66a00cb0a87133b2912ba5c1b2a02190  qtwayland-everywhere-src-$QT.tar.xz
+EOF
+
+wget -q https://libsdl.org/release/$SDL.tar.gz \
+https://github.com/NixOS/patchelf/releases/download/"$PATCHELF_VERSION/patchelf-$PATCHELF_VERSION.tar.bz2" \
+https://download.qt.io/official_releases/qt/"${QT%.*}/$QT/submodules/qtbase-everywhere-src-$QT.tar.xz" \
+https://download.qt.io/official_releases/qt/"${QT%.*}/$QT/submodules/qtsvg-everywhere-src-$QT.tar.xz" \
+https://download.qt.io/official_releases/qt/"${QT%.*}/$QT/submodules/qttools-everywhere-src-$QT.tar.xz" \
+https://download.qt.io/official_releases/qt/"${QT%.*}/$QT/submodules/qtwayland-everywhere-src-$QT.tar.xz" \
+https://download.qt.io/official_releases/qt/"${QT%.*}/$QT/submodules/qtdeclarative-everywhere-src-$QT.tar.xz" \
+https://download.qt.io/official_releases/qt/"${QT%.*}/$QT/submodules/qttranslations-everywhere-src-$QT.tar.xz" 
+sha256sum -c SHASUMS
+
+#find sha256sums
+sha256sum *.tar*
+
+tar -xf "$SDL.tar.gz"
+
+cd "$SDL"
+
+mkdir build && cd build
+
+cmake -DCMAKE_INSTALL_PREFIX="$INSTALLDIR" \
+-DCMAKE_BUILD_TYPE=MinSizeRel \
+-DCMAKE_PREFIX_PATH="$INSTALLDIR" \
+-GNinja \
+..
+
+ninja -j$(nproc) && ninja install
+
+cd ../..
+
+tar xf patchelf-$PATCHELF_VERSION.tar.bz2
+
+cd patchelf-$PATCHELF_VERSION
+
+./configure --prefix "$INSTALLDIR"
+make -j$(nproc) && make install
+cd ..
+
+    for f in qt*.tar*; do 
+        mkdir "${f/-*/}"
+        tar -xf "$f" -C "${f/-*/}" --strip-components=1; 
+    done
+
+
+	echo "Installing Qt Base..."
+	cd "qtbase"
+	cmake -B build -DCMAKE_PREFIX_PATH="$INSTALLDIR" -GNinja -DCMAKE_INSTALL_PREFIX="$INSTALLDIR" -DCMAKE_BUILD_TYPE=Release -DFEATURE_optimize_size=ON -DFEATURE_framework=OFF -DFEATURE_icu=OFF -DFEATURE_printsupport=OFF -DFEATURE_sql=OFF
```

Comment:
Why `-DFEATURE_framework=OFF`?
Why -`DFEATURE_icu=OFF`? (will this break unicode conversion?)
Why `-DFEATURE_optimize_size=ON`?

Reply:
These are pulled from the Mac build script, so I can tell you my reasoning for that
- `-DFEATURE_framework` is Mac specific, but I prefer dylibs there for bundling
- ICU is larger than the entire rest of the application combined, so I'd like to avoid it if possible.  I don't think we currently use it, but I'd expect it would be useful for things like splitting strings on unicode graphemes, locale-correct case conversions, etc.  I assume Windows currently builds with this off as well, since it doesn't ship with a 40mb ICU dll.  I doubt it'll break UTF-16 ↔︎ UTF-8 conversion, but it could break more obscure codecs if we used any (I hope we don't).
- I enabled optimize size because our dependencies tend to stay out of our hot paths, so I didn't think we needed `-O3`.  Saves a lot on SDL, though I don't know how much it saves on Qt.